apps/astarte_realm_management_api/priv/static/astarte_realm_management_api.yaml

```diff
@@ -464,6 +464,33 @@ paths:
           $ref: '#/components/responses/TriggerDeliveryPolicyCurrentlyUsedError'
         '500':
           description: Internal Server Error.
+  '/{realm_name}/devices/{device_id}':
+    parameters:
+      - $ref: '#/components/parameters/Realm'
+      - $ref: '#/components/parameters/DeviceID'
+    delete:
+      tags:
+        - device
+      summary: Delete device
+      description: >-
+        Deletes an existing device with a given `device_id`.
+        Device deletion is performed asynchronously.
```

Comment:
Maybe specify that, since deletion is asynchronous, a 204 response is not a guarantee that the device is deleted straightaway.